code/modules/mob/hear_say.dm

```diff
@@ -115,12 +115,6 @@
 
 	if(vname)
 		speaker_name = vname
-
-	if(istype(speaker, /mob/living/carbon/human))
-		var/mob/living/carbon/human/H = speaker
-		if(H.voice)
-			speaker_name = H.voice
```

Comment:
That check's already made in the radio's signal code. Basically, voice is updating on every life() and determines the voice of the character. By default real_name, unless masked, then name, unless they have a voicechanger/mimic, then the name of that.